tests/Baseline/spicy.optimization.unused-functions/log

```diff
@@ -1,4 +1,22 @@
 ### BTest baseline data generated by btest-diff. Do not edit. Use "btest -U/-u" to update. Requires BTest >= 0.63.
+[debug/global-optimizer] disabling feature 'supports_sinks' of type 'foo::B' since it is not used
+[debug/global-optimizer] disabling feature 'supports_sinks' of type 'foo::D' since it is not used
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__B%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__B%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__B%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__D%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__D%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__D%supports_filters' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__F%is_filter' at ''
+[debug/global-optimizer] inlining constant 'foo::__feat%foo__F%is_filter' at ''
+[debug/global-optimizer] removing conditional code around '' since the condition is always true
+[debug/global-optimizer] removing conditional code around '' since the condition is always true
+[debug/global-optimizer] removing conditional code around '' since the condition is always true
+[debug/global-optimizer] removing conditional code around '' since the condition is always true
+[debug/global-optimizer] removing conditional code around '' since the condition is always true
+[debug/global-optimizer] removing conditional code around '' since the condition is always true
+[debug/global-optimizer] removing conditional code around '' since the condition is always true
+[debug/global-optimizer] removing conditional code around '' since the condition is always true
```

Comment:
There's a lot of churn in these logs since we always run all optimizer passes. This makes it hard to see which pass actually lead to which behavior.

Since this PR introduces a baseclass for optimizer passes adding a way to explicitly enable certain passes from the outside shouldn't be too hard. For that we could e.g., look at an environment variable `HILTI_OPTIMIZER_PASSES` containing a comma-separated list of passes; if the env var is set we'd only enable to given passes.